### PR TITLE
Fix selects

### PR DESCRIPTION
**Pull request by Spacelocust**
## Description

Please include a summary of the changes and the related issue. Please also include relevant motivation and context. List any dependencies that are required for this change.

Closes #(issue) | No issue linked.

## Type of change

Please add any relevant information to the type of change you've made.

-   [x] Bug fix (non-breaking change which fixes an issue)
-   [ ] New feature (non-breaking change which adds functionality)
-   [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
-   [ ] This change requires a documentation update (or is a documentation update)
-   [x] HIGH IMPORTANCE (requires a fast merging)

## Has This Been Tested?

Please describe the tests that you ran to verify your changes. It is way more likely for a PR to be merged and approved with successful tests.

-   [x] Yes, I ran tests
-   [ ] No, I did not include any test
-   [ ] No test is needed for this change

### CODE REVIEW

File: front/app/lib/components/form/custom/FieldInputArray.tsx [hunk 1]
```diff
@@ -67,47 +83,57 @@ export function FieldInputArray<T extends FormData>({
...
                     control={control}
+                    render={({ field, fieldState: { error } }) => {
+                      const fieldId = id ?? field.name;
+                      const ariaLabelledBy = `${fieldId}-label`;
+                      const helpId = help ? `${fieldId}-help` : undefined;
+                      const errorId = error?.message ? `${fieldId}-error` : undefined;
+                      return (
+                        <>
+                          <LabelField
+                            id={ariaLabelledBy}
```
Comment (YummYume) [inline front/app/lib/components/form/custom/FieldInputArray.tsx]:
```suggestion
                            htmlFor={fieldId}
```

File: front/app/lib/components/form/custom/FieldInputArray.tsx [hunk 2]
```diff
@@ -67,47 +83,57 @@ export function FieldInputArray<T extends FormData>({
...
-                            {t(error.message, { ns: 'validators' })}
-                          </Text>
-                        )}
-                      </>
-                    )}
                     name={`${name}.${index}.${fieldInput.name}` as Path<T>}
                     control={control}
+                    render={({ field, fieldState: { error } }) => {
+                      const fieldId = id ?? field.name;
+                      const ariaLabelledBy = `${fieldId}-label`;
```
Comment (YummYume) [inline front/app/lib/components/form/custom/FieldInputArray.tsx]:
```suggestion
```

File: front/app/lib/components/form/custom/FieldMultiSelect.tsx [hunk 3]
```diff
@@ -54,21 +58,27 @@ export function FieldMultiSelect<T extends FormData>({
           }
           return (
             <>
+              <Text as="label" htmlFor={name} className={hideLabel ? 'sr-only' : undefined}>
```
Comment (YummYume) [inline front/app/lib/components/form/custom/FieldMultiSelect.tsx]:
```suggestion
              <Text as="label" id={ariaLabelledBy} htmlFor={name} className={hideLabel ? 'sr-only' : undefined}>
```

File: front/app/lib/components/form/custom/FieldMultiSelect.tsx [hunk 4]
```diff
@@ -54,21 +58,27 @@ export function FieldMultiSelect<T extends FormData>({
...
                 {...register(name)}
                 {...rest}
-                id={field.name}
+                id={fieldId}
+                aria-labelledby={ariaLabelledBy}
+                aria-describedby={helpId}
+                aria-errormessage={errorId}
+                aria-invalid={!!errorId}
                 key={`field_multi_select_key_${JSON.stringify(options)}`}
                 isMulti={true}
```
Comment (YummYume) [inline front/app/lib/components/form/custom/FieldMultiSelect.tsx:73]:
```suggestion
                isMulti={true}
```

ça serait bien qu'on puisse choisir si on veut du multi ou pas

File: front/app/lib/components/form/custom/FieldMultiSelect.tsx (hunk 3 above)
Comment (YummYume) [inline front/app/lib/components/form/custom/FieldMultiSelect.tsx]: LabelField ?

File: front/app/lib/components/form/custom/FieldMultiSelect.tsx (hunk 4 above)
Comment (Spacelocust) [reply front/app/lib/components/form/custom/FieldMultiSelect.tsx:73]: C'est ce que j'avais fait au début, mais le code n'était pas ouf

**Review by YummYume — APPROVED** [review]

**Pull request merged**